BambooTracker/tick_counter.cpp

```diff
@@ -58,7 +61,12 @@ void TickCounter::setSpeed(int speed)
 {
 	defStepSize_ = speed;
 	updateTickDifference();
-	tickDiffSum_ = 0;
+	// I choose to not set tickDiffSum_ to 0.
+	// This is because in BambooTracker and FamiTracker,
+	// changing speed does not reset tempo,
+	// so I decided to not interfere with the tempo accumulator either.
+	// However not resetting tickDiffSum_ has the downside that
+	// you can't reset the tempo accumulator mid-song using a speed command.
```

Comment:
Does this technique have a real-world use in a module or is this a strictly technical observation of behind-the-scenes behaviour? Have you seen it being used anywhere - and would this PR therefore break any existing modules?

Reply:
Currently, every tempo effect and every speed effect instantly reset the tempo accumulator(?). And one other person and I believe that *tempo* effects causing a delay is a bug which prevents smooth tempo transitions.

I don't know if there are any modules that depend on speed effects resetting the tempo accumulator. This will only affect modules which change speed using `F0x/F1x` after the tempo is changed away from 150 (or whatever equivalent value if the tick rate is changed from 60 Hz) using global tempo or `Fxx`. Even then, it should only have a minor effect, causing notes to be played 1 tick sooner or later (per row with a Fxx) at most. (I suspect that tempos above and below 150 have a different effect, because they result in `tickDiffSum_` having different signs, and `static_cast<int>(tickDiffSum_)` rounds towards 0.)

IMO, a bigger concern is fixing how each row with a `Fxx` has a full 1-tick delay. I think this is clearly a bugfix, but it might change how modules sound.

Note: the tempo system isn't really an "accumulator" like the N-SPC/Squaresoft SPC tempo systems are. It's implemented strangely internally, where the nominal speed and the tempo offset from the nominal speed are stored separately; maybe this can be changed to be more clear. What can't be changed is how BambooTracker computes the instantaneous duration of each row upfront, in order to release notes 1 tick before a new note triggers.

Reply:
I see, thanks for the thorough explanation.